### DISCUSSION TITLE
I want to get rid of the mxstep step.

### DISCUSSION THREAD

**Discussion by syeoplee** [pybamm-team/PyBaMM#3816] [Q&A] [answered]
Hello.

I am a student who is studying pybamm hard.

While writing the code, the result is always unfortunate due to mxstep steps.

I'm inquiring like this because I want some advice.

I post the code, plot, and error message together. This code only increased the discharge speed and it's all the same.

Thank you!

![first try](https://github.com/pybamm-team/PyBaMM/assets/159870255/ffb458f5-a586-482c-91de-4efc05cd5454)
At t = 277.328 and h = 2.39568e-14, the corrector convergence failed repeatedly or with |h| = hmin.
At t = 97.095, , mxstep steps taken before reaching tout.
At t = 97.095, , mxstep steps taken before reaching tout.
At t = 97.0951, , mxstep steps taken before reaching tout.
At t = 374.589, , mxstep steps taken before reaching tout.
At t = 134.584, , mxstep steps taken before reaching tout.
At t = 134.588, , mxstep steps taken before reaching tout.

```
import pybamm
import matplotlib.pyplot as plt

model = pybamm.lithium_ion.DFN()

experiment = pybamm.Experiment(["Discharge at 1C until 3.0 V"])
            
sim = pybamm.Simulation(
    model, experiment=experiment, parameter_values=pybamm.ParameterValues("Ai2020")
)
sol = sim.solve()

solution = sim.solution

x1 = solution["Discharge capacity [A.h]"]
V1 = solution["Voltage [V]"]

x1_dat = x1.entries
V1_dat = V1.entries

plt.plot(x1_dat,V1_dat)


experiment = pybamm.Experiment(["Discharge at 2C until 3.0 V"])
            
sim = pybamm.Simulation(
    model, experiment=experiment, parameter_values=pybamm.ParameterValues("Ai2020")
)
sol = sim.solve()

solution = sim.solution

x2 = solution["Discharge capacity [A.h]"]
V2 = solution["Voltage [V]"]

x2_dat = x2.entries
V2_dat = V2.entries

plt.plot(x2_dat,V2_dat)


experiment = pybamm.Experiment(["Discharge at 3C until 3.0 V"])
            
sim = pybamm.Simulation(
    model, experiment=experiment, parameter_values=pybamm.ParameterValues("Ai2020")
)
sol = sim.solve()

solution = sim.solution

x3 = solution["Discharge capacity [A.h]"]
V3 = solution["Voltage [V]"]

x3_dat = x3.entries
V3_dat = V3.entries

plt.plot(x3_dat,V3_dat)


experiment = pybamm.Experiment(["Discharge at 4C until 3.0 V"])
            
sim = pybamm.Simulation(
    model, experiment=experiment, parameter_values=pybamm.ParameterValues("Ai2020")
)
sol = sim.solve()

solution = sim.solution

x4 = solution["Discharge capacity [A.h]"]
V4 = solution["Voltage [V]"]

x4_dat = x4.entries
V4_dat = V4.entries

plt.plot(x4_dat,V4_dat)

experiment = pybamm.Experiment(["Discharge at 5C until 3.0 V"])
            
sim = pybamm.Simulation(
    model, experiment=experiment, parameter_values=pybamm.ParameterValues("Ai2020")
)
sol = sim.solve()

solution = sim.solution

x5 = solution["Discharge capacity [A.h]"]
V5 = solution["Voltage [V]"]

x5_dat = x5.entries
V5_dat = V5.entries

plt.plot(x5_dat,V5_dat)
```

**Comment by brosaplanella** [comment, marked as answer]
This is not an error. Despite exceeding the maximum number of steps, PyBaMM tries a few times with before giving up and throwing an actual error message. The message comes straight from the solver so we can't disabled it, unfortunately. But as long as this is the only message you get and the simulation reaches the end without throwing other errors it is fine.
